.github/workflows/ubuntu-cuda.yml

```diff
@@ -56,65 +49,57 @@ jobs:
           maximize_ubuntu_github_actions_build_space
       - name: Docker build
         run: |
-          docker build \
-            --build-arg BASE_IMAGE=${{ env.BASE_IMAGE }} \
-            --build-arg DEVELOPER_BUILD=${{ env.DEVELOPER_BUILD }} \
-            --build-arg BUILD_COMMON_CUDA_ARCHS=${{ env.BUILD_COMMON_CUDA_ARCHS }} \
-            --build-arg CCACHE_TAR_NAME=${{ env.CCACHE_TAR_NAME }} \
-            --build-arg CMAKE_VERSION=${{ env.CMAKE_VERSION }} \
-            --build-arg CCACHE_VERSION=${{ env.CCACHE_VERSION }} \
-            --build-arg PYTHON_VERSION=${{ env.PYTHON_VERSION }} \
-            -t open3d-ubuntu-cuda-ci:latest \
-            -f .github/workflows/Dockerfile.ubuntu-cuda .
-      - name: Extract wheel from Docker
-        run: |
-          mkdir -p ${GITHUB_WORKSPACE}/artifacts
-          docker run --rm \
-            -v ${GITHUB_WORKSPACE}/artifacts:/opt/mount \
-            open3d-ubuntu-cuda-ci:latest \
-            bash -c "cp /root/Open3D/build/lib/python_package/pip_package/open3d*.whl                /opt/mount && \
-                     cp /root/Open3D/build/lib/python_package/conda_package/linux-64/open3d*.tar.bz2 /opt/mount && \
-                     cp /${{ env.CCACHE_TAR_NAME }}.tar.gz                                           /opt/mount"
-          sudo chown -R $(id -u):$(id -g) ${GITHUB_WORKSPACE}/artifacts
-          ls -alh ${GITHUB_WORKSPACE}/artifacts
-          PIP_PKG_NAME="$(basename ${GITHUB_WORKSPACE}/artifacts/open3d*.whl)"
+          if   [ "${{ env.PYTHON_VERSION }}" = "3.6" ] && [ "${{ env.DEVELOPER_BUILD }}" = "ON"  ]; then
+            .github/workflows/docker_build.sh cuda_wheel_py36_dev
+          elif [ "${{ env.PYTHON_VERSION }}" = "3.7" ] && [ "${{ env.DEVELOPER_BUILD }}" = "ON"  ]; then
+            .github/workflows/docker_build.sh cuda_wheel_py37_dev
+          elif [ "${{ env.PYTHON_VERSION }}" = "3.8" ] && [ "${{ env.DEVELOPER_BUILD }}" = "ON"  ]; then
+            .github/workflows/docker_build.sh cuda_wheel_py38_dev
+          elif [ "${{ env.PYTHON_VERSION }}" = "3.9" ] && [ "${{ env.DEVELOPER_BUILD }}" = "ON"  ]; then
+            .github/workflows/docker_build.sh cuda_wheel_py39_dev
+          elif [ "${{ env.PYTHON_VERSION }}" = "3.6" ] && [ "${{ env.DEVELOPER_BUILD }}" = "OFF" ]; then
+            .github/workflows/docker_build.sh cuda_wheel_py36
+          elif [ "${{ env.PYTHON_VERSION }}" = "3.7" ] && [ "${{ env.DEVELOPER_BUILD }}" = "OFF" ]; then
+            .github/workflows/docker_build.sh cuda_wheel_py37
+          elif [ "${{ env.PYTHON_VERSION }}" = "3.8" ] && [ "${{ env.DEVELOPER_BUILD }}" = "OFF" ]; then
+            .github/workflows/docker_build.sh cuda_wheel_py38
+          elif [ "${{ env.PYTHON_VERSION }}" = "3.9" ] && [ "${{ env.DEVELOPER_BUILD }}" = "OFF" ]; then
+            .github/workflows/docker_build.sh cuda_wheel_py39
+          fi
```

Comment:
[nit] Slightly shorter if we separate the PYTHON_VERSION and DEVELOPER_BUILD checking:
```
if [ "${{ env.DEVELOPER_BUILD }}" = "ON" ]; then
    DEV="_dev"
else
    DEV=""
# Check PYTHON_VERSION and run
.github/workflows/docker_build.sh cuda_wheel_py37${DEV}
```

Reply:
Yes, I considered that initially as well. We could also go further and remove the `.` in python version number, i.e. `3.6` -> `36`, the code will look like:

```bash
PY_FLAG="_${${{ env.PYTHON_VERSION }}//.}"
if [ "${{ env.DEVELOPER_BUILD }}" = "ON" ]; then
  DEV_FLAG="_dev"
else
  DEV_FLAG=""
fi
DOCKER_OPTION="cuda_wheel${PY_FLAG}${DEV_FLAG}"
.github/workflows/docker_build.sh ${DOCKER_OPTION}
```

However, with the more verbose approach:
```bash
if   [ "${{ env.PYTHON_VERSION }}" = "3.6" ] && [ "${{ env.DEVELOPER_BUILD }}" = "ON"  ]; then
  .github/workflows/docker_build.sh cuda_wheel_py36_dev
elif [ "${{ env.PYTHON_VERSION }}" = "3.7" ] && [ "${{ env.DEVELOPER_BUILD }}" = "ON"  ]; then
  .github/workflows/docker_build.sh cuda_wheel_py37_dev
elif [ "${{ env.PYTHON_VERSION }}" = "3.8" ] && [ "${{ env.DEVELOPER_BUILD }}" = "ON"  ]; then
  .github/workflows/docker_build.sh cuda_wheel_py38_dev
elif ...
  ...
fi
```

it is more explicit and a developer can directly copy the command:
```bash
.github/workflows/docker_build.sh cuda_wheel_py36_dev
```
to run on the local machine.